### PR TITLE
[edn] Clear auto cmd FIFOs after edn disable

### LINKED ISSUE #19653: [edn] Discussion on Generate/Reseed cmd FIFOs behavior when disabling EDN
### Description

Currently the Reseed_cmd and the Generate_cmd FIFOs get cleared whenever the CTRL. CMD_FIFO_RST flag is set to true or when the edn_main_sm generates a done pulse. A done_pulse is generated on 3 different occasions:

1. Then edn_main_sm transitions from the BootPulse to the BootDone state (in boot_req_mode)
2. The edn_main_sm transitions from the AutoDispatch to the Idle state (when auto_req_mode is disabled)
3. The edn_main_sm transitions from the Idle state to the sw_port_mode state (when boot/auto_req_mode are both disabled)

The FIFOs aren't cleared however, when the EDN is disabled. E.g. the transition from the AutoDispatch to the Idle state could also happen when the EDN is disabled. In that case, the FIFOs aren't cleared. Even though it has the same initial and final state as point 2.

We should discuss if this is the intended behavior. And if not, when do we want to clear the FIFOs?

The results of this discussion should be added to the documentation.

FYI: @vogelpi @andreaskurth

### PR DESCRIPTION
**Pull request by h-filali**
This PR clears the auto cmd FIFOs after the edn is disabled.
For more precise information check out the individual commit messages.

This PR resolves #19653

### CODE REVIEW

**Comment by h-filali** [comment]
To all reviewers, please take a look at the commit message of d33e0d9c65c1e7ed5c9794beaf84864edbe27aac

File: hw/ip/edn/doc/theory_of_operation.md
Comment (rswarbrick) [inline hw/ip/edn/doc/theory_of_operation.md:74]: I'd consider splitting this sentence, which might make it easier to follow. Maybe "... to disable the auto request mode. Do this by configuring the CTRL register to deassert the AUTO_REQ_MODE or EDN_ENABLE field."

(I struggled to phrase this in a way that didn't sound like AUTO_REQ_MODE was a mode, but EDN_ENABLE was a field!)

File: hw/ip/edn/doc/theory_of_operation.md
Comment (rswarbrick) [inline hw/ip/edn/doc/theory_of_operation.md:77]: I think this might be easier to understand if the sentence gets split up a bit? I *think* I interpreted this right. Another possible meaning is that the FIFOs are also reset whenever the EDN gets disabled (in which case my suggested text below is wrong!)

```
The generate and reseed FIFOs are reset whenever the `SwPortMode` state is entered while the EDN is disabled.
They are also reset on completion of the boot sequence.
```

**Review by rswarbrick — COMMENTED** [review]
The DV code change looks sensible to me. A couple of nitty comments about the documentation change.

On the RTL side, it looks reasonable but I don't know enough about the design. I'm going to add @marnovandermaas and @vogelpi in the hope that one of them has more confidence :-)

File: hw/ip/edn/doc/theory_of_operation.md
Comment (h-filali) [reply hw/ip/edn/doc/theory_of_operation.md:74]: Agreed, thanks! I'll change it in my next push.

File: hw/ip/edn/doc/theory_of_operation.md
Comment (h-filali) [reply hw/ip/edn/doc/theory_of_operation.md:77]: Okay in that case it's probably best I change this. Thanks for reviewing.

**Comment by h-filali** [comment]
> The DV code change looks sensible to me. A couple of nitty comments about the documentation change.
> 
> On the RTL side, it looks reasonable but I don't know enough about the design. I'm going to add @marnovandermaas and @vogelpi in the hope that one of them has more confidence :-)

Thanks @rswarbrick for your review. The documentation should hopefully be easier to understand now.

**Comment by a-will** [comment]
FYI, I believe this PR will need to park here until the master branch is open for RTL changes. The current phase of our branching strategy would need `master` to be kept in line with `earlgrey_es`.

CC @msfschaffner for confirmation / more info

**Comment by h-filali** [comment]
@a-will, yes that's completely fine thanks. I'm just cleaning up some issues that are assigned to me.

File: hw/ip/edn/rtl/edn_main_sm.sv
Comment (vogelpi) [inline hw/ip/edn/rtl/edn_main_sm.sv:242]: The RTL change looks good to me. However, we can't merge RTL changes right now due to the focus SiVal right now.

Thanks for the detailed commit message. I think also the third case is not really an issue because when re-enabling EDN, we anyway expect software to rewrite the `max_num_between_reseeds` register, i.e., the  counter is expected to be re-initialized anyway. Moreover, I think it would be a bad idea to let the counter state survive disable/enable cycles. Or in other words, this PR also is an improvement with respect to handling that counter.

**Review by vogelpi — CHANGES_REQUESTED** [review]
The RTL change looks good to me! So do the DV changes. I have some nits regarding the documentation.

I am marking this PR as `Reqest changes` to block merging because at this very moment we should not merge RTL changes into master to not block SiVal / ES bring up.

File: hw/ip/edn/doc/theory_of_operation.md
Comment (h-filali) [reply hw/ip/edn/doc/theory_of_operation.md:74]: Good point. I'll change this back to the initial sentence.

File: hw/ip/edn/doc/theory_of_operation.md
Comment (h-filali) [reply hw/ip/edn/doc/theory_of_operation.md:77]: Thank you! I'll change this in the next push.

File: hw/ip/edn/rtl/edn_main_sm.sv
Comment (h-filali) [reply hw/ip/edn/rtl/edn_main_sm.sv:242]: Thanks! I agree with your assessment as well.

File: hw/ip/edn/doc/theory_of_operation.md
Comment (vogelpi) [reply hw/ip/edn/doc/theory_of_operation.md]: Thanks!

**Review by vogelpi — APPROVED** [review]
Approving as the RTL blocker is now gone. Will merge this once CI passes.

**Pull request merged**
